voluseg/_tools/parameter_dictionary.py

```diff
@@ -1,39 +1,129 @@
+        Cell diameter in microns (default is 6.0).
+    dir_ants : str, optional
+        Path to the ANTs directory (default is an empty string).
+    dir_input : str, optional
+        Path to input directories, separate multiple directories with ';' (default is an empty string).
+    dir_output : str, optional
+        Path to the output directory (default is an empty string).
+    dir_transform : str, optional
+        Path to the transform directory (default is an empty string).
+    ds : int, optional
+        Spatial coarse-graining in x-y dimension (default is 2).
+    planes_pad : int, optional
+        Number of planes to pad the volume with for robust registration (default is 0).
+    planes_packed : bool, optional
+        Packed planes in each volume, for single plane imaging with packed planes (default is False).
+    parallel_clean : bool, optional
+        Parallelization of final cleaning (True is fast but memory-intensive, default is True).
+    parallel_volume : bool, optional
+        Parallelization of mean-volume computation (True is fast but memory-intensive, default is True).
+    save_volume : bool, optional
+        Save registered volumes after segmentation, True keeps a copy of the volumes (default is False).
+    type_timepoints : str, optional
+        Type of timepoints to use for cell detection: 'dff', 'periodic' or 'custom' (default is 'dff').
+    type_mask : str, optional
+        Type of volume averaging for the mask: 'mean', 'geomean' or 'max' (default is 'geomean').
+    timepoints : int, optional
+        Number ('dff', 'periodic') or vector ('custom') of timepoints for segmentation (default is 1000).
+    f_hipass : float, optional
+        Frequency (Hz) for high-pass filtering of cell timeseries (default is 0).
+    f_volume : float, optional
+        Imaging frequency in Hz (default is 2.0).
+    n_cells_block : int, optional
+        Number of cells in a block. Small number is fast but can lead to blocky output (default is 316).
+    n_colors : int, optional
+        Number of brain colors (2 in two-color volumes, default is 1).
+    res_x : float, optional
+        X resolution in microns (default is 0.40625).
+    res_y : float, optional
+        Y resolution in microns (default is 0.40625).
+    res_z : float, optional
+        Z resolution in microns (default is 5.0).
+    t_baseline : int, optional
+        Interval for baseline calculation in seconds (default is 300).
+    t_section : float, optional
+        Exposure time in seconds for slice acquisition (default is 0.01).
+    thr_mask : float, optional
+        Threshold for volume mask: 0 < thr <= 1 (probability) or thr > 1 (intensity) (default is 0.5).
 
     Returns
     -------
     dict
-        Parameters dictionary with default values.
+        A dictionary of validated parameters with their default values.
     """
+    # Validate and parse the input parameters using the Pydantic model
+    params = ParametersModel(
```

Comment:
I think the call to ParametersModel is unnecessarily duplicating all the parameters. Can you check if the following replacement will work instead? (I have also renamed `params` to `parameters` for consistency.)

```
parameters = ParametersModel(**locals())
return parameters.dict()
```

Reply:
good, I included your suggestion but with one extra verification step to make sure no unwanted local variables would be included to instantiate the model